src/communication/network_interface_sr.c

```diff
@@ -7268,6 +7269,385 @@ slogtb_dump_trantable (THREAD_ENTRY * thread_p, unsigned int rid, char *request,
   db_private_free_and_init (thread_p, buffer);
 }
 
+/*
+ * smmon_get_server_info -
+ *
+ * return:
+ *
+ *   rid(in):
+ *   request(in):
+ *   reqlen(in):
+ */
+void
+smmon_get_server_info (THREAD_ENTRY * thread_p, unsigned int rid, char *request, int reqlen)
+{
+  char *buffer, *ptr;
+  int size = 0;
+  OR_ALIGNED_BUF (OR_INT_SIZE + OR_INT_SIZE) a_reply;
+  char *reply = OR_ALIGNED_BUF_START (a_reply);
+  int error = NO_ERROR;
+  MMON_SERVER_INFO server_info;
+
+  mmon_aggregate_server_info (server_info);
+
+  size += or_packed_string_length (server_info.name, NULL);
+  // Size of total_mem_usage
+  size += OR_INT64_SIZE;
+
+  buffer = (char *) db_private_alloc (thread_p, size);
+
+  if (buffer == NULL)
+    {
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY, 1, size);
+      error = ER_OUT_OF_VIRTUAL_MEMORY;
+    }
+  else
+    {
+      ptr = or_pack_string (buffer, server_info.name);
+      ptr = or_pack_int64 (ptr, server_info.total_mem_usage);
+    }
+
+  if (error != NO_ERROR)
+    {
+      ptr = or_pack_int (reply, 0);
+      ptr = or_pack_int (ptr, error);
+      css_send_data_to_client (thread_p->conn_entry, rid, reply, OR_ALIGNED_BUF_SIZE (a_reply));
+    }
+  else
+    {
+      ptr = or_pack_int (reply, size);
+      ptr = or_pack_int (ptr, error);
+      css_send_data_to_client (thread_p->conn_entry, rid, reply, OR_ALIGNED_BUF_SIZE (a_reply));
+      css_send_data_to_client (thread_p->conn_entry, rid, buffer, size);
```

Comment:
`css_send_reply_and_data_to_client` seems used for this kind of job. Have you considered it?

Reply:
Done.